staging/src/k8s.io/legacy-cloud-providers/azure/azure_controller_common.go

```diff
@@ -177,15 +182,19 @@ func (c *controllerCommon) DetachDisk(diskName, diskURI string, nodeName types.N
 
 	// make the lock here as small as possible
 	diskOpMutex.LockKey(instanceid)
+	c.diskAttachDetachMap.Store(strings.ToLower(diskURI), "")
 	resp, err := vmset.DetachDisk(diskName, diskURI, nodeName)
+	c.diskAttachDetachMap.Delete(strings.ToLower(diskURI))
```

Comment:
Can we please add a few UTs to cover this scenario ?

Reply:
we don't have a good fake UT framework to simulate this now.